### DISCUSSION TITLE
How can I get the size of a Widget?

### DISCUSSION THREAD

**Discussion by NickNick** [emilk/egui#879] [Q&A]
I was working with Grid and had to specify the row size, but found it hard to figure out what the size of a button is. Does it make sense to make the size that will be used for drawing (this one right: https://docs.rs/egui/0.15.0/src/egui/widgets/button.rs.html#168 ?) public in some way or should I go about this differently? 

Thanks!

**Comment by parasyte** [comment]
I think in an ideal world, I would like to have a method like `Widget::desired_size` that can provide this information for all widgets. It may even be useful as part of #843

**Reply by NickNick** [reply]
Yeah, something like that. I think some widgets do know their size (and some don't), so maybe `desired_size -> Option<Vec2>` or something? Or maybe an enum for `None/Some(Exact)/Some(Min, Max)`? Can I help with this?
